### PR TITLE
docs(headless): add React samples for headless result-list

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/KIT-382

Created a new pull request since there's been more changes this time.

Notably, introducing `fieldsToInclude` made the implementation of a functional component considerably more complicated since its fields need to be included in the controller's options.

### CODE REVIEW

File: packages/samples/headless-react/src/components/result-list/result-list.class.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    return (
+      <div>
+        <ul style={{textAlign: 'left'}}>
+          {this.state.results.map((result) => (
+            <li key={result.uniqueId}>
+              <article>
+                <h2>
+                  <a href={result.clickUri}>{result.title}</a>
```
Comment (olamothe) [inline packages/samples/headless-react/src/components/result-list/result-list.class.tsx:48]: One thing we absolutely need to add is an example on how to properly hook the click events.

It is very important that our clients do this ;)

File: packages/samples/headless-react/src/components/result-list/result-list.class.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                </h2>
+                <table>
```
Comment (olamothe) [inline packages/samples/headless-react/src/components/result-list/result-list.class.tsx:50]: To simplify the example, maybe we could do without a table ?  
Event if it's a bit less visually appealing.

File: packages/samples/headless-react/src/components/result-list/result-list.fn.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  return (
+    <div>
+      <ul style={{textAlign: 'left'}}>
+        {state.results.map((result) => (
+          <li key={result.uniqueId}>
+            <article>
+              <h2>
+                <a href={result.clickUri}>{result.title}</a>
+              </h2>
+              <table>
```
Comment (olamothe) [inline packages/samples/headless-react/src/components/result-list/result-list.fn.tsx:34]: Same comment for the table.

File: packages/samples/headless-react/src/components/result-list/result-list.fn.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  return (
+    <div>
+      <ul style={{textAlign: 'left'}}>
+        {state.results.map((result) => (
+          <li key={result.uniqueId}>
+            <article>
+              <h2>
+                <a href={result.clickUri}>{result.title}</a>
```
Comment (olamothe) [inline packages/samples/headless-react/src/components/result-list/result-list.fn.tsx:32]: Same comment for the example for the analytics event on click

**Review by olamothe — APPROVED** [review]

File: packages/samples/headless-react/src/components/result-list/result-list.class.tsx (hunk 1 above)
Comment (btaillon) [reply packages/samples/headless-react/src/components/result-list/result-list.class.tsx:48]: Good point. You mean `bindLogDocumentOpenOnResult(...)` in `result-utils.ts`?

File: packages/samples/headless-react/src/components/result-list/result-list.fn.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+              </h2>
+              <table>
+                <tbody>
+                  {Object.keys(props.fieldsToInclude).map(
```
Comment (samisayegh) [inline packages/samples/headless-react/src/components/result-list/result-list.fn.tsx:36]: Let's remove the `fieldsToInclude` prop and not render the fields. It makes the sample too complex. Just showing a clickable title (line 32) and the excerpt is more than enough.

**Review by samisayegh — APPROVED** [review]

File: packages/samples/headless-react/src/components/result-list/result-list.fn.tsx (hunk 5 above)
Comment (btaillon) [reply packages/samples/headless-react/src/components/result-list/result-list.fn.tsx:36]: For only the functional component or both?

Comment (samisayegh) [reply packages/samples/headless-react/src/components/result-list/result-list.fn.tsx:36]: I would do it for both to keep the samples consistent. If someone toggles between the two, I think they should get the same rendered output

**Pull request closed without merging**